corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -224,13 +225,49 @@ def form_exists(form_id, domain=None):
             return result.form_exists
 
     @staticmethod
-    @transaction.atomic
```

Comment:
Why did you remove the atomic decorator? Would it make sense to move this to the block where forms are deleted?